lib/dedupe.js

```diff
@@ -142,8 +142,12 @@ function hoistChildren_ (tree, diff, seen, next) {
         [andComputeMetadata(tree)]
       ], done)
     }
-    var hoistTo = earliestInstallable(tree, tree.parent, child.package, log)
-    if (hoistTo) {
+    // find a location where it's installable
+    // child is marked as removed so it's ignored when finding a location
+    child.removed = true
+    var hoistTo = earliestInstallable(tree, tree, child.package, log)
+    child.removed = false
```

Comment:
I think we'd rather add an argument here than patch a child statefully like this. This otherwise makes sense.